pgfinder/pgio.py

```diff
@@ -248,7 +248,8 @@ def dataframe_to_csv(save_filepath: Union[str, Path], filename: str, output_data
     output_dataframe: pd.DataFrame
         Pandas Dataframe to write to csv
     """
-
+    if "diff_ppm" not in output_dataframe.columns:
+        output_dataframe = _calculate_ppm_delta(df=output_dataframe)
```

Comment:
An alternative place to stick this might be line 61 of `find_pg.py`? I guess it's down to whether one feels it's part of the output or part of the analysis... Would perhaps avoid the `if`...

Reply:
Might also help with test coverage, but that's not a big deal, here.

Reply:
Thanks, I added the `if` clause as I was going round in circles.

I first added the call to `dataframe_to_csv()` then when I added it to `dataframe_to_csv_metadata()` I encountered an error because the column already existed (since `dataframe_to_csv_metadata()` is called prior to `dataframe_to_csv()`).

I haven't understood the logic for saving the dataframe to string in memory and where/when its then written out. Having been getting more familiar with typing in Python whilst the function always returns a string (just realised the docstring is wrong since `Path()` is wrapped in `str()`) its seems strange that it could either be a string to a file path or a CSV in memory.